### ISSUE TITLE
Mac binaries are reported as "damaged"

### ISSUE THREAD

**Issue by nonoche2** [Ecks1337/RyuSAK#37]
both Mac binaries for Ryusak 1.6 bring up an error alert when launched saying that the apps are damaged and can't be opened. (By the way I believe any Mac app is a "portable" app so you shouldn't need two different versions)

**To Reproduce**
1. download either Mac binaries on a Mac
2. launch them
3. see error

- OS: macOS Ventura 13.0.1
- Version 1.6.0

**Comment by Diadriu** [comment]
It's not a RyuSAK issue, the application is simply not signed.
You need to disable gatekeeper, I posted a fix.

**Comment by nonoche2** [comment]
that's the thing, unsigned apps don't give this "damaged" error but an "unsigned" error (which allows to launch them either by right-clicking their icon and selecting "open" the first time, or allowing their launch the first time in the system settings). Disabling Gatekeeper altogether is neither recommanded nor easily done on the latest version of macOS.

The "damaged" error means that the app has been tampered with, that's a different issue

**Comment by ningvw520** [comment]
encounter same error, showed the app was damaged instead of unsigned. mac os 13.0.1, m1 max.

**Comment by Diadriu** [comment]
> that's

I believe it's a gatekeeper error, I posted a fix how to disable gatekeeper for the app

**Comment by nonoche2** [comment]
yes, you are correct, this error is reported by GateKeeper, however, this isn't the usual alert Gatekeeper throws for merely unsigned apps, and one does not need to disable gatekeeper altogether to launch them. When a regularly unsigned app is blocked by Gatekeeper, this is the alert it throws :
<img width="372" alt="package01" src="https://user-images.githubusercontent.com/5378758/205503567-fc9abfb1-d62b-40f8-bb09-aed6c74c98aa.png">

In that case, you only have to allow the application once in system settings, or right click the app and select "open", then you get this alert once : 
![package02](https://user-images.githubusercontent.com/5378758/205503721-4a1affd7-50e1-47aa-a45f-c256bc7d9c04.png)

When we get the "damaged" alert, it means gatekeeper thinks the binary has been tampered with, and the only way to launch it is to disable gatekeeper altogether. This isn't necessary to launch an unsigned app and I believe there's something wrong in the app complilation setting, I'll look into it and will let you know

**Comment by nonoche2** [comment]
Did you get the application notarized with Apple? It is now mandatory, but you only need a free Apple developer account to do so

https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution

**Comment by Diadriu** [comment]
Thanks for the clarification. Hope there is a way for you to disable gatekeeper for this application on the newest macos.

**Comment by nonoche2** [comment]
I'm not sure I have been clear enough yet ^^ There shouldn't be any need to disable gatekeeper to launch and unsigned application. Anyone can perfectly launch an unsigned application with gatekeeper enabled. But we can't do so with Ryusak because it's reported as hacked by gatekeeper, not just unsigned. There's something wrong with the way you built the app

**Comment by TomKowalczuk** [comment]
Andrew Tsai found a way to fix it 
just type this in the terminal 
`xattr -cr "RyuSAK.app location"`

**Comment by M0REKZ** [comment]
yeah, this command would allow you to run the app, but wont really fix it

**Comment by Diskutant** [comment]
> Andrew Tsai found a way to fix it
> just type this in the terminal
> `xattr -cr "RyuSAK.app location"`

the correct command is
`xattr -cr "RyuSAK.app"`

**Comment by therobertbula** [comment]
I'm still getting the broken message, and the command doesn't do anything
Do I have to have the app somewhere in specific?

**Comment by TomKowalczuk** [comment]
Make sure to put 
`xattr -cr /Applications/RyuSAK.app`
(if that's where you put it)

**Comment by Diadriu** [comment]
You can try this line first
sudo spctl --master-disable

Then enter this line (change path to app if needed):
xattr -cr /Applications/RyuSAK.app

**Comment by Ecks1337** [comment]
> (By the way I believe any Mac app is a "portable" app so you shouldn't need two different versions)

@nonoche2 The portable version is different because it stores all its files within its folder rather than in the applications data directory

**Comment by nonoche2** [comment]
oh I see, thanks for the clarification. I was under the impression that "portable" meant "can be used on any computer from a USB key without an installer", but if that implies keeping the settings in one place it makes sense. I can't say I have seen this kind of setup for any other app on macOS though.

**Comment by Ecks1337** [comment]
After looking into it a bit, it seems that I'd need to use xcode in order to sign and notarise the app. I don't own a Mac so i see no way to resolve this other than to accept #34 as the official solution

**Comment by nonoche2** [comment]
could this help?

https://github.com/zkmkarlsruhe/mac-dist-helper

https://github.com/zkmkarlsruhe/mac-dist-helper-examples

**Comment by xycainoff** [comment]
i get same error. 
“RyuSAK” is damaged and can’t be opened. You should move it to the Bin.

**Comment by BigEmperor26** [comment]
`xattr -cr /Applications/RyuSAK.app`

after you do that right click and then open the app

**Comment by OrigamingWasTaken** [comment]
Hello, I tried doing `xattr -cr /Applications/RyuSAK.app`, however, it says that the `-r` option is invalid.
```
❯ xattr -cr /Applications/RyuSAK.app


option -r not recognized

usage: xattr [-slz] file [file ...]
       xattr -p [-slz] attr_name file [file ...]
       xattr -w [-sz] attr_name attr_value file [file ...]
       xattr -d [-s] attr_name file [file ...]
       xattr -c [-s] file [file ...]

The first form lists the names of all xattrs on the given file(s).
The second form (-p) prints the value of the xattr attr_name.
The third form (-w) sets the value of the xattr attr_name to attr_value.
The fourth form (-d) deletes the xattr attr_name.
The fifth form (-c) deletes (clears) all xattrs.

options:
  -h: print this help
  -s: act on symbolic links themselves rather than their targets
  -l: print long format (attr_name: attr_value)
  -z: compress or decompress (if compressed) attribute value in zip format```

**Comment by OrigamingWasTaken** [comment]
Sorry for bothering! For anyone with the same problem, do: `pip uninstall xattr`
The original xattr installed with pip was overwriting Apple's implementation of it. So the `-r` option was gone.
